lang/python38/files/patch-threadid-older-systems.diff

```diff
@@ -0,0 +1,23 @@
+--- Python/thread_pthread.h.orig	2020-03-07 12:42:06.000000000 -0800
++++ Python/thread_pthread.h	2020-03-07 12:48:22.000000000 -0800
+@@ -325,9 +325,19 @@
+ {
+     if (!initialized)
+         PyThread_init_thread();
+ #ifdef __APPLE__
+     uint64_t native_id;
+-    (void) pthread_threadid_np(NULL, &native_id);
++#if MAC_OS_X_VERSION_MAX_ALLOWED < 1060
++    native_id = pthread_mach_thread_np(pthread_self());
```

Comment:
After looking into this some more, I'm not sure anymore that this is the best solution. The entire definition of `PyThread_get_thread_native_id` is actually inside an `#ifdef PY_HAVE_THREAD_NATIVE_ID` block, i. e. CPython supports not having that function at all. It might be a better idea to change the definition of `PY_HAVE_THREAD_NATIVE_ID` (in Include/pythread.h) so that it is only defined on Snow Leopard and newer (which have the required `pthread_threadid_np` function).

The documentation for `PyThread_get_thread_native_id()` (or rather its Python counterparts, [`threading.get_native_id()`](https://docs.python.org/3/library/threading.html#threading.get_native_id) and [`threading.Thread.native_id`](https://docs.python.org/3/library/threading.html#threading.Thread.native_id)) specifically points out that the returned value is unique *system-wide*. If I'm remembering the discussion from the issue tracker correctly, this isn't guaranteed for `pthread_mach_thread_np`.

Since `PyThread_get_thread_native_id` and friends were only added in Python 3.8, I assume that most code will still use the older `PyThread_get_thread_ident`, at least as a fallback. Any code that unconditionally uses the new `PyThread_get_thread_native_id` probably depends on the returned ID being unique system-wide, so I think it would be better to have such code fail fast by not providing the new function if we can't implement it correctly.